### PR TITLE
canvas, it works.

### PR DESCRIPTION
**Pull request by CodeTorso**
# Refactor and Enhance Canvas Component

## Overview
This PR introduces significant changes to the canvas component, including refactoring and the addition of new features. The primary focus is on improving the user experience and enhancing the functionality of the canvas-based application.

## Changes
- **Refactoring:**
    - Removed several components and their associated logic, including `page.tsx`, `search&create.tsx`, `thinkPad.tsx`, and `thinkPads.tsx`. This suggests a potential shift in the application's architecture or functionality.
    - Refactored the `Canvas` and `TldrDrawComponent` components to improve code organization and readability.
    - Moved the `handleDragOver` function and related logic to the `Canvas` component for better drag and drop event handling.
    - Introduced the `DragContext.Provider` to manage drag and drop state across canvas components.
    - Simplified the `TldrDrawComponent` by moving remote snapshot loading logic to a separate effect.
    - Optimized the `TldrDrawComponent` by using the `memo` higher-order component to memoize the component and prevent unnecessary re-renders.
    - Moved the `createAssetFromUrl` function to the `lib/createAssetUrl.tsx` file for improved code organization.
    - Refactored the `TldrDrawComponent` to use the `useRect` hook for retrieving the unique identifier of the canvas.
    - Optimized the `TldrDrawComponent` by using the `memo` higher-order component to memoize the component and prevent unnecessary re-renders.
- **New Features:**
    - Implemented a new `DropZone` component to handle file drops on the canvas.
    - Added support for creating embeds from URLs using the `createEmbedsFromUrl` function.
    - Integrated the `twitterCardUtil` and `textCardUtil` shape utilities to provide specialized handling for Twitter cards and text cards.
    - Registered external asset and content handlers to the `TldrDraw` editor, enabling the creation of assets and embeds from URLs.
    - Introduced a new "Text Card" custom node type to the canvas component, allowing users to add text-based content with editable content and additional information.
    - Implemented a new `ResizableLayout` component for dynamic resizing of the left panel and main canvas area, enhancing user flexibility.
    - Integrated the `SidePanel` component into the new layout, improving organization and interaction with the canvas.
    - Added a new `Search` function to the `SidePanel` component, allowing users to input queries and fetch relevant sources from an API.
    - Implemented drag-and-drop functionality for cards within the `SidePanel`, enabling users to visually interact with content.
    - Added context for drag-and-drop functionality to manage drag state across components.
    - Dynamically adjusted the height of the `Card` component based on its content using the `useMeasure` hook from `react-use-measure`.
- **Other Changes:**
    - Removed the `apps/web/app/api/unfurlSite/route.tsx` file, as it appears to be unrelated to the canvas functionality.
    - Updated the `apps/web/app/actions/doers.tsx` file to redirect to the ThinkPad canvas page.
    - Updated the `apps/web/app/actions/doers.tsx` file to use a more robust regular expression for URL matching in the `typeDecider` function.
    - Added a new `unfurlSite` function to the `apps/web/app/actions/fetchers.tsx` file to fetch website metadata.
    - Added a new `canvasStyles.css` file for styling the ThinkPad canvas.

> ✨ Generated with love by Kaizen ❤️

<details>
<summary>Original Description</summary>
None
</details>

### CODE REVIEW

**Comment by Dhravya** [comment]
huge

File: apps/web/app/api/unfirlsite/route.ts
Comment (aryasaatvik) [inline apps/web/app/api/unfirlsite/route.ts:1]: Can DEV_IMAGES be removed now? https://github.com/CodeTorso/supermemory/blob/canvas/apps/web/env.d.ts#L7

Comment (CodeTorso) [reply apps/web/app/api/unfirlsite/route.ts:1]: I still need a r2 bucket if someone pastes like images on canvas, but i guess we can use the STORAGE r2 bucket for that.

Comment (CodeTorso) [reply apps/web/app/api/unfirlsite/route.ts:1]: so yes

**Review by Dhravya — APPROVED** [review]

**Comment by CodeTorso** [comment]
hmmm it broke :(

**Pull request merged**